Patch based on,

commit d190c9a41e3b4a3121af5ed9ac759c3d5a277ee1
Author: Steven R. Loomis <srloomis@us.ibm.com>
Date:   Wed Feb 7 16:42:21 2018 -0800

    src: add "icu::" prefix before ICU symbols

    In ICU 61.x, icu4c will no longer put its declarations in the global namespace.
    Everything will be in the "icu::" namespace (or icu_60:: in the linker).
    Prepare for this.
    https://ssl.icu-project.org/trac/ticket/13460

Index: node-v4.9.1/tools/icu/iculslocs.cc
===================================================================
--- node-v4.9.1.orig/tools/icu/iculslocs.cc
+++ node-v4.9.1/tools/icu/iculslocs.cc
@@ -64,7 +64,7 @@ int VERBOSE = 0;
 #define RES_INDEX "res_index"
 #define INSTALLEDLOCALES "InstalledLocales"
 
-CharString packageName;
+icu::CharString packageName;
 const char* locale = RES_INDEX;  // locale referring to our index
 
 void usage() {
@@ -147,7 +147,7 @@ int localeExists(const char* loc, UBool*
   if (VERBOSE > 1) {
     u_printf("Trying to open %s:%s\n", packageName.data(), loc);
   }
-  LocalUResourceBundlePointer aResource(
+  icu::LocalUResourceBundlePointer aResource(
       ures_openDirect(packageName.data(), loc, &status));
   *exists = FALSE;
   if (U_SUCCESS(status)) {
@@ -189,11 +189,11 @@ void printIndent(const LocalUFILEPointer
  * @return 0 for OK, 1 for err
  */
 int dumpAllButInstalledLocales(int lev,
-                               LocalUResourceBundlePointer* bund,
+                               icu::LocalUResourceBundlePointer* bund,
                                LocalUFILEPointer* bf,
                                UErrorCode* status) {
   ures_resetIterator(bund->getAlias());
-  LocalUResourceBundlePointer t;
+  icu::LocalUResourceBundlePointer t;
   while (U_SUCCESS(*status) && ures_hasNext(bund->getAlias())) {
     t.adoptInstead(ures_getNextResource(bund->getAlias(), t.orphan(), status));
     ASSERT_SUCCESS(status, "while processing table");
@@ -255,10 +255,10 @@ int list(const char* toBundle) {
     u_printf("\"locale\": %s\n", locale);
   }
 
-  LocalUResourceBundlePointer bund(
+  icu::LocalUResourceBundlePointer bund(
       ures_openDirect(packageName.data(), locale, &status));
   ASSERT_SUCCESS(&status, "while opening the bundle");
-  LocalUResourceBundlePointer installedLocales(
+  icu::LocalUResourceBundlePointer installedLocales(
       ures_getByKey(bund.getAlias(), INSTALLEDLOCALES, NULL, &status));
   ASSERT_SUCCESS(&status, "while fetching installed locales");
 
@@ -294,7 +294,7 @@ int list(const char* toBundle) {
   }
 
   // OK, now list them.
-  LocalUResourceBundlePointer subkey;
+  icu::LocalUResourceBundlePointer subkey;
 
   int validCount = 0;
   for (int32_t i = 0; i < count; i++) {
